tests/quantization/quanto/test_quanto.py

```diff
@@ -0,0 +1,177 @@
+        model.to("cuda")
+
+        assert (model.get_memory_footprint() / 1024**3) < self.expected_memory_use_in_gb
+        for name, module in model.named_modules():
+            if isinstance(module, torch.nn.Linear):
+                if name in model._keep_in_fp32_modules:
+                    assert module.weight.dtype == torch.float32
+        self.model_cls._keep_in_fp32_modules = _keep_in_fp32_modules
+
+    def test_dtype_assignment(self):
+        model = self.model_cls.from_pretrained(**self.get_dummy_model_init_kwargs())
+        assert (model.get_memory_footprint() / 1024**3) < self.expected_memory_use_in_gb
+
+        with self.assertRaises(ValueError):
+            # Tries with a `dtype`
+            model.to(torch.float16)
+
+        with self.assertRaises(ValueError):
+            # Tries with a `device` and `dtype`
+            model.to(device="cuda:0", dtype=torch.float16)
+
+        with self.assertRaises(ValueError):
+            # Tries with a cast
+            model.float()
+
+        with self.assertRaises(ValueError):
+            # Tries with a cast
+            model.half()
+
+        # This should work
+        model.to("cuda")
+
+
+class FluxTransformerQuantoMixin(QuantoBaseTesterMixin):
+    model_id = "hf-internal-testing/tiny-flux-transformer"
+    model_cls = FluxTransformer2DModel
+    torch_dtype = torch.bfloat16
+    keep_in_fp32_module = "proj_out"
+
+    def get_dummy_inputs(self):
+        return {
+            "hidden_states": torch.randn((1, 4096, 64), generator=torch.Generator("cpu").manual_seed(0)).to(
+                torch_device, self.torch_dtype
+            ),
+            "encoder_hidden_states": torch.randn(
+                (1, 512, 4096),
+                generator=torch.Generator("cpu").manual_seed(0),
+            ).to(torch_device, self.torch_dtype),
+            "pooled_projections": torch.randn(
+                (1, 768),
+                generator=torch.Generator("cpu").manual_seed(0),
+            ).to(torch_device, self.torch_dtype),
+            "timestep": torch.tensor([1]).to(torch_device, self.torch_dtype),
+            "img_ids": torch.randn((4096, 3), generator=torch.Generator("cpu").manual_seed(0)).to(
+                torch_device, self.torch_dtype
+            ),
+            "txt_ids": torch.randn((512, 3), generator=torch.Generator("cpu").manual_seed(0)).to(
+                torch_device, self.torch_dtype
+            ),
+            "guidance": torch.tensor([3.5]).to(torch_device, self.torch_dtype),
+        }
+
+
+class FluxTransformerFloat8(FluxTransformerQuantoMixin, unittest.TestCase):
+    expected_memory_use_in_gb = 10
+
+    def get_dummy_init_kwargs(self):
+        return {"weights": "float8"}
+
+
+class FluxTransformerInt8(FluxTransformerQuantoMixin, unittest.TestCase):
+    expected_memory_use_in_gb = 10
+
+    def get_dummy_init_kwargs(self):
+        return {"weights": "int8"}
+
+    def test_torch_compile(self):
```

Comment:
Why is this only a part of `FluxTransformerInt8`. This is could be a part of `FluxTransformerQuantoMixin`, no?